core/src/main/java/org/apache/carbondata/core/datamap/status/DataMapStatusManager.java

```diff
@@ -95,7 +95,9 @@ public static void disableAllLazyDataMaps(CarbonTable table) throws IOException
         DataMapStoreManager.getInstance().getDataMapSchemasOfTable(table);
     List<DataMapSchema> dataMapToBeDisabled = new ArrayList<>(allDataMapSchemas.size());
     for (DataMapSchema dataMap : allDataMapSchemas) {
-      if (dataMap.isLazy()) {
+      // TODO all non datamaps like MV is now supports only lazy. Once the support is made the
+      // following check can be removed.
+      if (dataMap.isLazy() || !dataMap.isIndexDataMap()) {
```

Comment:
But pre-aggregate datamap supports immediate load with main table.
`!dataMap.isIndexDataMap()` returns true for pre-aggregate datamap, right?

Reply:
pre aggregate datamap flow is entirely different, it does not come here.